tests/kernel/interrupt/src/regular_isr.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <interrupt_util.h>
+
+/*
+ * Other arch has already been tested in testcase of gen_isr_table,
+ * so we only test x86 series here.
+ */
+#if defined(CONFIG_X86)
+
+#define TEST_IRQ_LINE_1	27
+#define TEST_IRQ_LINE_2	28
+
+#define TRIGGER_IRQ_LINE_1	64
+#define TRIGGER_IRQ_LINE_2	65
+
+#define TEST_IRQ_PRIO	2
+
+
+volatile uint32_t reg_int_executed[2];
+
+void isr_comm(const void *param)
+{
+	int choice = POINTER_TO_INT(param);
+
+	switch (choice) {
+	case TEST_IRQ_LINE_1:
+		reg_int_executed[0]++;
+	break;
+	case TEST_IRQ_LINE_2:
+		reg_int_executed[1]++;
+	break;
+
+	default:
+	break;
+	}
+}
+
+/**
+ * @brief Test regular interrupt
+ *
+ * @details Validate regular interrupt works as expected.
+ * - Register two regular interrupt at build time.
+ * - Trigger interrupt and check if isr handler has executed or not.
+ * - Also check irq_enable and irq_disable works.
+ *
+ * @ingroup kernel_interrupt_tests
+ *
+ * @see IRQ_CONNECT(), irq_enable(), irq_disable(),
+ * irq_unlock(),
+ */
+void test_isr_regular(void)
+{
+	int trig_vec1, trig_vec2;
+
+	IRQ_CONNECT(TEST_IRQ_LINE_1, TEST_IRQ_PRIO, isr_comm, (void *)TEST_IRQ_LINE_1, 0);
+	IRQ_CONNECT(TEST_IRQ_LINE_2, TEST_IRQ_PRIO, isr_comm, (void *)TEST_IRQ_LINE_2, 0);
+
+#if defined(CONFIG_X86)
+	trig_vec1 = Z_IRQ_TO_INTERRUPT_VECTOR(TEST_IRQ_LINE_1);
+	trig_vec2 = Z_IRQ_TO_INTERRUPT_VECTOR(TEST_IRQ_LINE_2);
+#elif defined(CONFIG_ARCH_POSIX)
+	trig_vec1 = TRIGGER_IRQ_LINE_1;
+	trig_vec2 = TRIGGER_IRQ_LINE_2;
+#endif
+
+	/* verify the target triggering vector is correct */
+	zassert_equal(trig_vec1, TRIGGER_IRQ_LINE_1,
```

Comment:
You know "trig_vec1" must be "64". Why do we have to call macro Z_IRQ_TO_INTERRUPT_VECTOR() at line 61?

Reply:
Hi, thanks for your review. @LeiW000 because here we want to check that hardcoded TRIGGER_IRQ_LINE 64 is the correct vector which is decided at build time. It indeed looks a little weird, but doing that can make sure the hardcoded vector here is correct or not.  Thanks!

(Or you might wonder why we have to hardcode TRIGGER_IRQ_LINE 64 here. There is a limitation that the x86 INT command only supports immediate numbers,  this means the input parameter of the following trigger_irq( ) must be known at build time, using variables as parameter will cause compilation failure.)

Reply:
Got it. Thanks.